writeaheadlog.go

```diff
@@ -133,6 +133,9 @@ func newWal(path string, deps dependencies) (u []Update, w *WAL, err error) {
 
 		// Recover WAL and return updates
 		updates, err := newWal.recoverWal(data)
+		if err != nil {
+			newWal.logFile.Close()
```

Comment:
It is probably safe to ignore the error here but maybe we could compose them for completeness. We might want to move the `ComposeError` function to a neutral repo though to avoid references to Sia.
```
if err != nil {
    err = build.ComposeErrors(err, newWal.logFile.Close())
}
```